data-raw/spawning-habitat-decay/spawning-decay.R

```diff
@@ -0,0 +1,413 @@
+    total_scaledown
+)
+
+AVG_flow_cfs_to_sed_cfd_final <- approxfun(
+  x = rating_curve$flow_cfs, 
+  y = rating_curve$sed_ft3_per_day_avg * 
+    DSMhabitat::gravel_size_to_prop_of_movement$avg_fraction * 
+    total_scaledown
+)
+
+MAX_flow_cfs_to_sed_cfd_final <- approxfun(
+  x = rating_curve$flow_cfs, 
+  y = rating_curve$sed_ft3_per_day_max * 
+    DSMhabitat::gravel_size_to_prop_of_movement$avg_fraction * 
+    total_scaledown
+)
+
+# Exceedance probs --------------------------------------
+dsm_flows <- DSMflow::flows_cfs$biop_itp_2018_2019 |> 
+  pivot_longer(cols = -date, names_to = "watershed", values_to = "flow_cfs")
+
+watersheds_with_decay <- names(which(DSMhabitat::watershed_decay_status))
+
+exceedance_curves <- map(watersheds_with_decay, function(w) {
+  d <- dsm_flows |> filter(watershed == w) |> 
+    mutate(cume_dist = dplyr::cume_dist(-flow_cfs)) |> 
+    arrange(desc(cume_dist))
+  
+  approxfun(x = d$cume_dist, y = d$flow_cfs)
+}) |> 
+  set_names(watersheds_with_decay)
+
+
+exceedance_curves$`Upper Sacramento River`(0.0569578) # ~18000
+
+upper_sac_exceedance_at_18k <- 0.0569578
+
+watershed_offsets <- map_dbl(fallRunDSM::watershed_labels, function(w) {
+  if (w %in% watersheds_with_decay) {
+    exceedance_curves[[w]](upper_sac_exceedance_at_18k)
+  } else {
+    NA_real_
+  }
+}) |> set_names(fallRunDSM::watershed_labels)
+
+watershed_spawning_decays <- map2(fallRunDSM::watershed_labels, watershed_offsets, function(w, x) {
+  
+  if (w %in% watersheds_with_decay) {
+    dsm_flows |>
+      filter(watershed == w, year(date) %in% 1979:2000) |> 
+      mutate(flow_adjusted = flow_cfs - x, 
+             decay_min = MIN_flow_cfs_to_sed_cfd_final(flow_adjusted), 
+             decay_avg = AVG_flow_cfs_to_sed_cfd_final(flow_adjusted), 
+             decay_max = MAX_flow_cfs_to_sed_cfd_final(flow_adjusted) 
+      ) |> 
+      pivot_longer(names_to = "decay_type", values_to = "decay_amount", -c(date, watershed, 
+                                                                           flow_cfs, flow_adjusted)) |> 
+      mutate(decay_cfd = ifelse(is.na(decay_amount), 0, decay_amount), 
+             decay_cfm = decay_cfd * days_in_month(month(date)),
+             decay_sqm = decay_cfm / 2, 
+             decay_acres_month = decay_sqm / 43560, 
+             decay_type = stringr::str_extract(decay_type, "min|avg|max")) |> 
+      select(date, watershed, flow_cfs, decay_type, decay_acres_month) 
+  } else {
+    dsm_flows |> 
+      filter(watershed == w, year(date) %in% 1979:2000) |> 
+      transmute(
+        date, watershed, flow_cfs, decay_type = "none", decay_acres_month = 0
+      )
+  }
+}) |> 
+  set_names(fallRunDSM::watershed_labels)
+
+usethis::use_data(watershed_spawning_decays, overwrite = TRUE)
+
+watershed_decays$`American River` |> 
+  ggplot(aes(date, decay_acres_month, color = decay_type)) + geom_line()
+
+watershed_decays$`American River` |> 
```

Comment:
When running the script I can't find the object `watershed_decays`. I think you may have renamed it watershed_spawning_decays.

Reply:
fixed